### PR TITLE
preconditioned CG (Diag Scale) and rescale parameters

### PR DESCRIPTION
**Pull request by k-ido**
- add supports for preconditioned SR-CG and rescaling variational parameters (only Slater and Proj)
ref: H.-H. Zhao, Phys. Rev. B 96, 085103 (2017)
- update documents and tests
- debug minor issues for initializations and freeing memory after mVMC calculations.

### CODE REVIEW

**Comment by tmisawa** [comment]
@k-ido 

Applied follow-up fixes in commit decd30d:

- Fixed `test/python/runtest_sr.py`:
  - check `result_cg` correctly after the second run
  - compare SR vs CG with `max(abs(diff))` instead of signed mean
  - fail fast on shape mismatch

- Fixed `RescaleSmat` edge cases:
  - added empty-range guard in `get_absmax()`
  - skip rescaling when `NProj <= 0` or `NProj >= NPara`
  - skip ratio calculation when projected max is effectively zero

These changes prevent false negatives in SR-CG tests and avoid boundary-condition issues in rescaling.

File: test/python/runtest_sr.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+if result_cg != 0:
+    sys.exit(result_cg)
+
+
+array_calc_cg = read_out("./output/zqp_opt.dat")[4:]
+
+result = 0
+if array_calc_sr.shape != array_calc_cg.shape:
+    sys.exit(-1)
+diff = np.abs(np.mean(array_calc_sr - array_calc_cg))
```
[the remark this answers is not in the document]
Comment (tmisawa) [reply test/python/runtest_sr.py]: Use mean absolute SR-CG difference in test
